### DISCUSSION TITLE
github-discussion-comments-for-jekyll-blog

### DISCUSSION THREAD

**Discussion by giscus** [bartoszgorka/bartoszgorka.github.io#112] [Announcements]
# github-discussion-comments-for-jekyll-blog

I introduced the possibility of commenting on posts directly on the page. With giscus, we can use Discussions on GitHub to store comments. In this simple way, you can add comments to your Jekyll blog!

https://bartoszgorka.com/github-discussion-comments-for-jekyll-blog

**Comment by patrykwozinski** [comment]
Wow, that's a brilliant functionality. Thanks for your post!

**Reply by bartoszgorka** [reply]
You're welcome! Very simple functionality to implement and gives great possibilities.

**Comment by ApolloFops** [comment]
This is pretty cool! It'll be pretty useful for my weblog (I was looking at [this other solution](https://dc25.github.io/myBlog/2017/06/24/using-github-comments-in-a-jekyll-blog.html) but it would take some styling to get it to look good and doesn't support replies to comments).

**Reply by bartoszgorka** [reply]
Simple solution and really fast to introduce 😄 Good to see working not only for me.

**Comment by seletz** [comment]
Thanks man!  Pretty cool!  I use the chirpy theme, which has built-in config options for this -- never looked at it, but now found your post by accident and [promptly enabled it](https://seletz.github.io/posts/giscus-comments)!

**Reply by bartoszgorka** [reply]
Thank you!

**Comment by stevenharman** [comment]
Wow, so happy to kick Disqus to the curb. It took all of a few minutes (mostly in understanding the configuration options) to get it set up on [my blog](https://stevenharman.net/so-we-have-a-memory-leak). Thank you! 🙌

**Reply by bartoszgorka** [reply]
Really nice blog! I will check it after work.

For me it was also great to kick Disqus because sharing a lot of details regarding each visitor and their policy was not great for me.

**Comment by JuliMillan** [comment]
This is great. Thank you!

**Reply by bartoszgorka** [reply]
Thank you!

**Comment by ebenezerrahul** [comment]
This looks great. thanks!

**Reply by bartoszgorka** [reply]
Thank you!

**Comment by fdatri** [comment]
thanks!

**Comment by an7tang** [comment]
Thank you for sharing. So helpful!!

**Reply by bartoszgorka** [reply]
Thank you for support! Simple solution working amazing 🚀

**Comment by knzai** [comment]
I was just thinking of implementing something like this for my blog, and thought "I bet it already exists huh."  But it looks like it does require a running server, and I want to figure out if I can do it all with client-side (well and freezing the existing comments to static in the GH Action whenever I republish) via the GH rest library.

So your post about utterances and giscus gives me some interesting jumping off points to explore, thanks.

**Reply by bartoszgorka** [reply]
Previously I used Disqus but due to their issue with privacy and processing data, I decided to check giscus and working amazing 🚀

**Reply by knzai** [reply]
For now I settled on [Mastodon comments](https://github.com/dpecos/mastodon-comments), since I already have a Mastodon account that I'm posting the blog posts to anyway, and it's easy to drop in for purely client-side use (though that means no graceful degradation, unless I wanna like fetch the results, put them into the static content, then on client side get the comments, only add the new ones, etc, etc).


But I'll definitely keep it in mind for some other more GH repo centric stuff.

**Comment by VadimNastoyashchy** [comment]
cool

**Reply by bartoszgorka** [reply]
Thanks!

**Comment by Ewewerban** [comment]
Very cool, thanks !

**Reply by bartoszgorka** [reply]
You're welcome!
